parser_library/test/compare_unordered_vectors.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2019 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#define EXPECT_VECTORS_EQ_UNORDERED(actual, expected)                                                                  \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        ASSERT_EQ(actual.size(), expected.size());                                                                     \
+        for (const auto& i : actual)                                                                                   \
+        {                                                                                                              \
+            auto it = std::find(expected.begin(), expected.end(), i);                                                  \
+            EXPECT_NE(it, expected.end()) << "Test returned unexpected value:" << i;                                   \
+        }                                                                                                              \
```

Comment:
this does not work for {1,1} and {1,2}

I suggest sorting if possible and then compare in order.

Also expected cannot be an expression, because it would be evaluated multiple times, so I suggest a helper functions.